sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/package.scala

```diff
@@ -29,12 +29,23 @@ package object analysis {
 
   /**
    * Resolver should return true if the first string refers to the same entity as the second string.
-   * For example, by using case insensitive equality.
+   * For example, by using case insensitive equality. Besides, Resolver also provides the ability
+   * to normalize the string according to its semantic.
    */
-  type Resolver = (String, String) => Boolean
+  trait Resolver {
+    def apply(a: String, b: String): Boolean
+    def apply(a: String): String
+  }
+
+  val caseInsensitiveResolution = new Resolver {
+    override def apply(a: String, b: String): Boolean = a.equalsIgnoreCase(b)
+    override def apply(a: String): String = a.toLowerCase // as Hive does
```

Comment:
I'd like keep the first `apply` as it was, because I don't want to impact a lots of existed code. I agree  we should rename the second `apply` => `normalize`.